src/robotFactory.js

```diff
@@ -1,10 +1,72 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: (coords.x) ? coords.x : 0,
```

Comment:
```suggestion
      x: coords.x ?? 0,
```
very useful operator (perfect in this case): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

Reply:
Thank you! I didn't know about that. It's really useful
But linter shows an error when i try to us it, therefore I'll keep my code as it is